### PR TITLE
Fave a zoop

### LINKED ISSUE #55: Fave a Zoop
- [ ] You must be an authenticated user to Fave a Zoop.
- [ ] An authenticated user Fave any Zoop EXCEPT one that they sent

### PR DESCRIPTION
**Pull request by dyazdani**
Closes #55 

Fave button disabled when not logged in:
Screen Shot 2023-12-02 at 22 18 53

After logging in:
Screen Shot 2023-12-02 at 22 19 33

After clicking the fave button of the top Zoop (changed from 2 to 3):
Screen Shot 2023-12-02 at 22 19 50

The fave shows up on the single Zoop page:
Screen Shot 2023-12-02 at 22 20 06

The fave button is disabled if you sent the Zoop:
Screen Shot 2023-12-02 at 22 23 40

https://github.com/dyazdani/zoop/assets/99094815/32c34c06-87dd-4b0f-b813-4bff7494b80a

### CODE REVIEW

File: client/components/FaveButton.tsx [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+        zoopId: zoop.id
+      })
+    }
   })
   
   return (
-      <Button variant="outlined" onClick={(e) => handleClick(e)}>
+      <Button 
+        variant="outlined"
+        disabled={!currentUser || currentUser.id === zoop.authorId}
```
Comment (paloobi) [inline client/components/FaveButton.tsx:37]: Might be wise to show some tooltip if disabled to explain why

**Review by paloobi — APPROVED** [review]
It would be nice to have some visual confirmation when you have faved a zoop. Could you make the star icon filled if you faved the zoop? (Maybe makes sense as a separate issue & PR)

File: client/components/FaveButton.tsx (hunk 1 above)
Comment (dyazdani) [reply client/components/FaveButton.tsx:37]: I'll add an issue

Comment (r-matias) [reply client/components/FaveButton.tsx:37]: o guard null aqui é intencional — esse método é chamado pelo job de correção que lista ids em batch, e entre a listagem e o processamento o enrollment pode ter sido removido. nesse contexto, ignorar silenciosamente é o comportamento esperado. um log poderia ser útil em outros cenários, mas não justifica a mudança agora.

**Pull request merged**